geomesa-accumulo/geomesa-accumulo-tools/src/main/scala/org/locationtech/geomesa/accumulo/tools/AccumuloRunner.scala

```diff
@@ -107,9 +114,33 @@ object AccumuloRunner extends Runner {
             "change the Zookeeper timeout by setting the system property 'instance.zookeeper.timeout'.")
 
           import scala.concurrent.duration._
-          Await.result(Future(HdfsZooInstance.getInstance().getInstanceName)(ExecutionContext.global),  lookupTime.millis)
+          lazy val ff: Either[Throwable, String] = try {
+            // This call not only provides the instance name but in situations where the instance
+            // name is provided, this allows us to check that we have access to Accumulo dependencies.
+            // This helps prevent errors later and allows us to provide more relevant error messages now.
+            val i = HdfsZooInstance.getInstance().getInstanceName
+            Right(i)
+          } catch {
+            case e: Throwable => Left(e)
+          }
+          val future = Await.result(Future(ff)(ExecutionContext.global),  lookupTime.millis)
+
+          future match {
+            case Right(instanceId) =>
+              if (p.instance == null) p.instance = instanceId
+              else if (instanceId != p.instance) Command.user.warn("The provided instance ID (--instance) does not " +
+                "match the instance ID of the resolved Accumulo instance.")
+            case Left(e) => throw e
```

Comment:
@aheyne it looks like it will throw an exception if it can't load `accumulo-site.xml` regardless of it -i is defined (that also aligns with what @jahhulbert-ccri reported). We want to be able to specify -i and not require `accumulo-site.xml` on the classpath. (`HdfsZooInstance.getInstance().getInstanceName` loads `accumulo-site.xml`)
It seems like loading the zookeepers separately is also pointless, we can just use `HdfsZooInstance.getInstance().getZookeepers`, which does the same thing.
Is there any situation where you would need only one of -i or -z? It seems like the flow should be:
1. load classOf[Connection], catch ClassNotFoundException, throw param exception for jars/classpath
2. If one or both of -i and -z not set? Load HdfsZooInstance, validate any that are set, set to the loaded values. if can't load, throw a ParamException
3. any other case, don't load HdfsZooInstance - we don't need `accumulo-site.xml` on the classpath